### ISSUE TITLE
Statsd plugin only sends counter rate not the actual count

### ISSUE THREAD

**Issue by jamtur01** [collectd/collectd#1282]
If I look at https://github.com/collectd/collectd/blob/master/src/statsd.c#L264 then it appears that the statsd plugin only sends the rate metric when sending a counter.

If I send:

```bash
for i in {1..1000000}; do echo "productiona.app1-web1.counter:$i|c" | nc -u -w0 127.0.0.1 8125; done
```

Then I'd expect to see in statsd:

```json
Flushing stats at  Sat Sep 26 2015 10:55:41 GMT-0400 (EDT)
{ counters:
   { 'statsd.bad_lines_seen': 0,
     'statsd.packets_received': 1247,
     'statsd.metrics_received': 1247,
     'productiona.app1-web1.counter': 26624697 },
  timers: {},
  gauges: { 'statsd.timestamp_lag': 0 },
  timer_data: {},
  counter_rates:
   { 'statsd.bad_lines_seen': 0,
     'statsd.packets_received': 124.7,
     'statsd.metrics_received': 124.7,
     'productiona.app1-web1.counter': 2662469.7 },
  sets: {},
  pctThreshold: [ 90 ] }
```

With the statsd plugin I see:

```json
:host app1-web1, :service statsd/derive-productiona.app1-web1.counter, :state ok, :description nil, :metric 2662469.7, :tags [collectd], :time 1443279411, :ttl 60.0, :ds_index 0, :ds_name value, :ds_type derive:rate, :type_instance productiona.app1-web1.counter, :type derive, :plugin statsd}
```

Which is just the rate and NOT the actual count. 

According to the StatsD docs for counters: `Statsd will send both the rate as well as the count at each flush.` (https://github.com/etsy/statsd/blob/master/docs/metric_types.md)

Am I missing something here?

**Comment by pyr** [comment]
If you use `StoreRates false` in your riemann configuration, the metrics emitted will be absolute counts, as requested initially. If for any reason you want to mix rates and absolute counters, the only way would be to have a gauge emitted as well.

More generally, collectd is moving away from counters and towards only rates, gauges and timers.
Can you let me know if the code works for you, then @octo can weigh-in on his view of #1311.

My take on is is that in "standard" read plugins this should be avoided. In this case, since there are cases where collectd is used as a vessel between different systems, it might make sense to forward the absolute counters. I'm thinking of the scenario which you seem to be working with:

```
<app> -> <local statsd> -> <local collectd> -> <global riemann>
```

**Comment by jamtur01** [comment]
The code works for me.

**Comment by mfournier** [comment]
I also agree with the reasoning. Even if it might seem a bit awkward from a collectd perspective, +1 for matching the behaviour of the reference statsd implementation.

Thanks for working on this @jamtur01 & @pyr !

**Comment by pyr** [comment]
Fixed by #1311
